chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -39,8 +40,7 @@ object Module {
     //   - unset readyForModuleConstr
     //   - reset whenDepth to 0
     //   - set currentClockAndReset
-    val m = bc.setRefs()
-    m._commands.prepend(DefInvalid(childSourceInfo, m.io.ref)) // init module outputs
+    val module: T = bc  // Module is actually invoked here
```

Comment:
Technically, "bc is actually evaluated here"
You are already well underway of the Module invocation.